File: webapp-net/Search/Data/SearchResultSet.cs
using System.Collections.Generic;

namespace Sdl.Web.Modules.Search.Data
{
    public class SearchResultSet
    {
        public int Hits { get; set; }

        public int Count { get; set; }

        public int StartIndex { get; set; }

        public IList<SearchResult> QueryResults { get; set; }
    }   
}

File: webapp-java/dxa-module-tridion-docs-search/src/main/java/com/sdl/dxa/modules/docs/search/model/SearchResultSet.java
package com.sdl.dxa.modules.docs.search.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

/**
 * SearchResultSet represents data structure which contains search results alongside with additional search data.
 */
@Data
public class SearchResultSet {

    @JsonProperty("Hits")
    private Integer hits;

    @JsonProperty("Count")
    private Integer count;

    @JsonProperty("StartIndex")
    private Integer startIndex;

    @JsonProperty("QueryResults")
    private List<SearchResult> queryResults;
}
